spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java

```diff
@@ -0,0 +1,259 @@
+    // save the model to a file in the output FileSet
+    JavaSparkContext sparkContext = context.getSparkContext();
+    FileSet outputFS = context.getDataset(config.fileSetName);
+    model.save(JavaSparkContext.toSparkContext(sparkContext),
+               outputFS.getBaseLocation().append(config.path).toURI().getPath());
+
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext sparkPluginContext) throws Exception {
+    // no-op; no need to do anything
+  }
+
+  /**
+   * Configuration for DecisionTreeTrainer.
+   */
+  public static class DecisionTreeTrainerConfig extends PluginConfig {
+
+    @Description("The name of the FileSet to save the model to.")
+    private final String fileSetName;
+
+    @Description("Path of the FileSet to save the model to.")
+    private final String path;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields to use for training. If empty, all the fields will be " +
+      "considered for training. Features to be used, must be from one of the following type: int, long, float or " +
+      "double. Both featuresToInclude and featuresToExclude fields cannot be specified.")
+    private final String featuresToInclude;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields to exclude when training. If empty, all the fields will be " +
+      "considered for training. Both featuresToInclude and featuresToExclude fields cannot be specified.")
+    private final String featuresToExclude;
+
+    @Nullable
+    @Description("Mapping of the feature to the cardinality of that feature; required for categorical features.")
+    private final String cardinalityMapping;
+
+    @Description("The field from which to get the prediction. It must be of type double.")
+    private final String labelField;
+
+    @Nullable
+    @Description("Maximum depth of the tree.\n For example, depth 0 means 1 leaf node; depth 1 means 1 internal node " +
+      "+ 2 leaf nodes. Default is 10.")
+    private Integer maxDepth;
+
+    @Nullable
+    @Description("Maximum number of bins used for splitting when discretizing continuous featuresToInclude. " +
+      "DecisionTree requires maxBins to be at least as large as the number of values in each categorical feature. " +
+      "Default is 100.")
+    private Integer maxBins;
+
+    public DecisionTreeTrainerConfig(String fileSetName, String path, String featuresToInclude,
+                                     String featuresToExclude, String cardinalityMapping, String labelField,
+                                     Integer maxDepth, Integer maxBins) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featuresToInclude = featuresToInclude;
+      this.featuresToExclude = featuresToExclude;
+      this.cardinalityMapping = cardinalityMapping;
+      this.labelField = labelField;
+      this.maxDepth = maxDepth;
+      this.maxBins = maxBins;
+    }
+
+    private void validate(Schema inputSchema) {
```

Comment:
a lot of this logic is actually the same for the trainer and the predictor. It is better to create a DecisionTreeConfig, or perhaps a FeatureConfig, that all the configs extend. That way the same logic can be used for all the ML trainer and predictor plugins.